lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/DBOAccessRequirementDAOImpl.java

```diff
@@ -187,176 +202,112 @@ public Long mapRow(ResultSet rs, int rowNum) throws SQLException {
-		return basicDao.getCount(DBOAccessRequirement.class);
+		List<DBOSubjectAccessRequirement> nars = namedJdbcTemplate.query(GET_SUBJECT_ACCESS_REQUIREMENT_SQL, param, subjectAccessRequirementRowMapper);
+		return AccessRequirementUtils.copyDBOSubjectsToDTOSubjects(nars);
 	}
 
-	@Override
-	public List<AccessRequirement> getAllAccessRequirementsForSubject(List<String> subjectIds, RestrictableObjectType type)  throws DatastoreException {
-		List<AccessRequirement>  dtos = new ArrayList<AccessRequirement>();
-		if (subjectIds.isEmpty()) return dtos;
-		List<Long> subjectIdsAsLong = new ArrayList<Long>();
-		for (String id: subjectIds) {
-			subjectIdsAsLong.add(KeyFactory.stringToKey(id));
-		}
+	private DBOAccessRequirement getAccessRequirementForUpdate(Long accessRequirementId) throws NotFoundException{
 		MapSqlParameterSource param = new MapSqlParameterSource();
-		param.addValue(COL_SUBJECT_ACCESS_REQUIREMENT_SUBJECT_ID, subjectIdsAsLong);
-		param.addValue(COL_SUBJECT_ACCESS_REQUIREMENT_SUBJECT_TYPE, type.name());
-		List<DBOAccessRequirement> dbos = namedJdbcTemplate.query(SELECT_FOR_SUBJECT_SQL, param, accessRequirementRowMapper);
-		for (DBOAccessRequirement dbo : dbos) {
-			AccessRequirement dto = AccessRequirementUtils.copyDboToDto(dbo, getSubjects(dbo.getId()));
-			dtos.add(dto);
+		param.addValue(COL_ACCESS_REQUIREMENT_ID, accessRequirementId);
+		try {
+			return namedJdbcTemplate.queryForObject(SELECT_FOR_UPDATE_SQL, param, accessRequirementRowMapper);
+		}catch (EmptyResultDataAccessException e) {
+			throw new NotFoundException("The resource you are attempting to access cannot be found");
 		}
-		return dtos;
 	}
 
 	@WriteTransactionReadCommitted
 	@Override
-	public <T extends AccessRequirement> T update(T dto) throws DatastoreException,
-			InvalidModelException,NotFoundException, ConflictingUpdateException {
-		MapSqlParameterSource param = new MapSqlParameterSource();
-		param.addValue(COL_ACCESS_REQUIREMENT_ID, dto.getId());
-		List<DBOAccessRequirement> ars = null;
-		try{
-			ars = namedJdbcTemplate.query(SELECT_FOR_UPDATE_SQL, param, new RowMapper<DBOAccessRequirement>() {
-				@Override
-				public DBOAccessRequirement mapRow(ResultSet rs, int rowNum)
-						throws SQLException {
-					DBOAccessRequirement ar = new DBOAccessRequirement();
-					ar.setCreatedBy(rs.getLong(COL_ACCESS_REQUIREMENT_CREATED_BY));
-					ar.setCreatedOn(rs.getLong(COL_ACCESS_REQUIREMENT_CREATED_ON));	
-					ar.seteTag(rs.getString(COL_ACCESS_REQUIREMENT_ETAG));
-					return ar;
-				}
-				
-			});
-		}catch (EmptyResultDataAccessException e) {
-			throw new NotFoundException("The resource you are attempting to access cannot be found");
-		}
-		if (ars.isEmpty()) {
-			throw new NotFoundException("The resource you are attempting to access cannot be found");			
+	public <T extends AccessRequirement> T update(T dto) {
+		DBOAccessRequirement current = getAccessRequirementForUpdate(dto.getId());
```

Comment:
Want to move as much of this logic to the manager as possible.  Includes adding a new dao method to get minimum info about the current version (for update).